board/safety/safety_volkswagen.h

```diff
@@ -43,17 +93,27 @@ static void volkswagen_mqb_init(int16_t param) {
   relay_malfunction = false;
   volkswagen_torque_msg = MSG_HCA_01;
   volkswagen_lane_msg = MSG_LDW_02;
+  gen_crc_lookup_table(0x2F, volkswagen_crc8_lut_8h2f);
 }
 
 static int volkswagen_mqb_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
 
   bool valid = addr_safety_check(to_push, volkswagen_mqb_rx_checks, VOLKSWAGEN_MQB_RX_CHECKS_LEN,
-                                 NULL, NULL, NULL);
+                                 volkswagen_get_checksum, volkswagen_mqb_compute_crc, volkswagen_get_counter);
 
   if (valid) {
     int bus = GET_BUS(to_push);
     int addr = GET_ADDR(to_push);
 
+    // Update in-motion state by sampling front wheel speeds
+    // Signal: ESP_19.ESP_VL_Radgeschw_02 (front left)
+    // Signal: ESP_19.ESP_VR_Radgeschw_02 (front right)
+    if ((bus == 0) && (addr == MSG_ESP_19)) {
+      int wheel_speed_fl = GET_BYTE(to_push, 4) | (GET_BYTE(to_push, 5) << 8);
+      int wheel_speed_fr = GET_BYTE(to_push, 6) | (GET_BYTE(to_push, 7) << 8);
+      volkswagen_moving = (wheel_speed_fl > 0) || (wheel_speed_fr > 0);
```

Comment:
this is not exactly the same as what we have in OP. In OP we check for standstill (speed < 0.1m/s). Here we should be less aggressive and perhaps just check that the average of the front wheel speeds > 0.3m/s to assign the moving bool to true.

Reply:
I will do that. It'll be a little bit because I'll want to buff the unit test to look on both sides of the grace threshold, and I have to leave here shortly.